src/providers/twitch.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+
+import { OAuth2Client, OAuth2ClientConfig } from "../../deps.ts";
+import type { WithRedirectUri, WithScope } from "./_types.ts";
+
+/**
+ * Creates an OAuth 2.0 client with Twitch as the provider.
+ *
+ * Requires `--allow-env[=TWITCH_CLIENT_ID,TWITCH_CLIENT_SECRET]` permissions and environment variables:
+ * 1. `TWITCH_CLIENT_ID`
+ * 2. `TWITCH_CLIENT_SECRET`
+ *
+ * @param additionalOAuth2ClientConfig Requires `redirectUri` and `defaults.scope` properties.
+ *
+ * @example
+ * ```ts
+ * import { createTwitchOAuth2Client } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
+ *
+ * const oauth2Client = createTwitchOAuth2Client({
+ *  redirectUri: "http://localhost:8000/callback",
+ *  defaults: {
+ *    scope: "moderator:read:followers"
+ *  }
+ * });
+ * ```
+ *
+ * @see {@link https://dev.twitch.tv/docs/authentication/}
+ */
+export function createTwitchOAuth2Client(
+  additionalOAuth2ClientConfig:
+    & Partial<OAuth2ClientConfig>
+    & WithRedirectUri
+    & WithScope,
+): OAuth2Client {
+  const clientId = Deno.env.get("TWITCH_CLIENT_ID")!;
+  const clientSecret = Deno.env.get("TWITCH_CLIENT_SECRET")!;
+
+  return new OAuth2Client({
+    clientId,
+    clientSecret,
+    authorizationEndpointUri: "https://id.twitch.tv/oauth2/authorize",
+    tokenUri: "https://id.twitch.tv/oauth2/token",
+    ...additionalOAuth2ClientConfig,
+    defaults: {
+      ...additionalOAuth2ClientConfig.defaults,
+      requestOptions: {
+        ...additionalOAuth2ClientConfig.defaults.requestOptions,
+        urlParams: {
+          ...additionalOAuth2ClientConfig.defaults.requestOptions?.urlParams,
+          client_id: clientId,
+          client_secret: clientSecret,
+        },
+      },
```

Comment:
Can you confirm whether things still work if you remove this block? It might not be needed.

Reply:
Yeah, I commented that out and get a 400 response code to the `token` endpoint. This is because Twitch requires the `client_id` and `client_secret` as query params to that endpoint: https://dev.twitch.tv/docs/authentication/getting-tokens-oauth/#use-the-authorization-code-to-get-a-token

Reply:
```
Error: Invalid token response: Server returned 400 and no error description was given
    at RefreshTokenGrant.getTokenResponseError (XXXX/deno-oauth2-client/src/grant_base.ts:174:14)
    at eventLoopTick (ext:core/01_core.js:181:11)
    at async RefreshTokenGrant.parseTokenResponse (XXXX/deno-oauth2-client/src/grant_base.ts:72:13)
    at async getSessionAccessToken (XXXX/deno_kv_oauth/src/get_session_access_token.ts:41:17)
    at async indexHandler (XXXX/deno_kv_oauth/demo.ts:76:7)
    at async handler (XXXX/deno_kv_oauth/demo.ts:110:14)
    at async ext:deno_http/00_serve.js:507:22 {
  response: Response {
  body: ReadableStream { locked: true },
  bodyUsed: true,
  headers: Headers {
  "access-control-allow-origin": "*",
  "content-length": "45",
  "content-type": "application/json",
  date: "Mon, 07 Aug 2023 19:21:52 GMT",
  server: "nginx",
  "x-ctxlog-logid": "1-64d14450-3da2c8ea7880cc1f43158d4a"
},
  ok: false,
  redirected: false,
  status: 400,
  statusText: "Bad Request",
  url: "https://id.twitch.tv/oauth2/token"
}
}
```

Reply:
Oh, I now recall that Twitch OAuth doesn't support PKCE. We found this out previously in #79. Hm... I'd rather not have the Twitch implementation slightly jerry-rigged to work with this module. Instead, in the README, we could note that Twitch is not supported for this reason. WDYT?

Reply:
Ah, makes sense. Yeah we can do that

Reply:
Sorry, I just saw your last comment. I've created #157 to supersede this PR. Please feel free to review that PR. Thanks, Mitch!

Reply:
All good!